modules/godot/databags/databag_timer.h

```diff
@@ -0,0 +1,121 @@
+﻿#pragma once
+
+#include "../../../databags/databag.h"
+#include "core/math/math_defs.h"
+#include "core/templates/local_vector.h"
+
+namespace godex {
+typedef uint32_t TimerIndex;
+typedef uint32_t TimerGeneration; // Change to a bitfield in a struct if more data is needed for timer and timerHandle, or reduce to int16 if it's needed only on one of the two types, we don't need the full int 32 for the generation. Might as well use it until we find another purpose for this memory.
+
+struct Timer {
+	Timer(uint64_t p_endTime, godex::TimerGeneration p_generation) :
+			endTime(p_endTime), generation(p_generation) {}
+	Timer() :
```

Comment:
If we can avoid using a contructor is better, so the the compiler can optimize the default and copy constructor.

Or just use the syntax: `Timer() = default;` and set `end_timer = 0;`